clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3091,6 +3104,58 @@ static void handleMaxGlobalWorkDimAttr(Sema &S, Decl *D,
                                                                      E);
 }
 
+SYCLIntelLoopFuseAttr *
+Sema::mergeSYCLIntelLoopFuseAttr(Decl *D, const SYCLIntelLoopFuseAttr &Attr,
+                                 Expr *E) {
+  if (checkAttrMutualExclusion<SYCLIntelLoopFuseIndependentAttr>(*this, D,
+                                                                 Attr))
+    return nullptr;
+
+  if (D->hasAttr<SYCLIntelLoopFuseAttr>())
```

Comment:
Right. The silent dropping is the concern I emailed you about. I wasn't really show how it is 'normally' handled. 

>>Should we only allow the attribute to be written on a function definition by forcing the attribute to be late-parsed and checking whether the Decl is a function definition rather than a function declaration

I THINK it shouldn't matter to callers but I've emailed the person who requested this support just to be sure? I'll take a look at allowing this only on definitions.

Reply:
The usual pattern is: the `mergeFooAttr()` function checks all of the merging constraints and diagnoses problematic situations like mutually exclusive attributes, duplicate attributes with different arguments, etc, and the `handleFooAttr()` function checks all of the basic constraints and diagnoses problematic situations like incorrect number of attribute arguments, wrong argument type, etc. and if everything looks okay, calls `mergeFooAttr()` and only adds the resulting attribute from the merge function if it's nonnull.

The availability attribute is an example of one that has somewhat complicated checking in both `handle` and `merge`. What you have in this patch is somewhat close, but the merge function is not diagnosing the duplicate attribute before dropping it, and the handle function is doing a bit of manual work rather than calling the merge function.

Reply:
Thanks for the explanation. I think I changed the handling to make it more like this.